### DISCUSSION TITLE
How to reverse lookup addr to name?

### DISCUSSION THREAD

**Discussion by 0xc0de4c0ffee** [namesys-eth/ccip2-eth-resources#7] [Ideas]
We've to think deep about this one.. 

?? route it separately like how contenthash is used in current resolver?

**Comment by sshmatrix** [comment]
Is `name` in CCIP2 reverse record for the address? If yes, then I guess we'll need to route it separately so that the Reverse Registrar can CCIP-Read it... Did I get the "problem" correctly?

**Reply by 0xc0de4c0ffee** [reply]
this
```
Specification
New instances of the Name Wrapper, Reverse Registrar, and .eth registrar controller have been deployed to mainnet at these addresses:

NameWrapper: 0xD4416b13d2b3a9aBae7AcD5D6C2BbDBE25686401
ReverseRegistrar: 0xa58E81fe9b61B5c3fE2AFD33CF304c454AbFc7Cb
ETHRegistrarController: 0x253553366Da8546fC250F225fe3d25d0C782303b
PublicResolver: 0x231b0Ee14048e9dCcD1d247744d114a4EB5E8E63
```
https://www.tally.xyz/gov/ens/proposal/42973781582803845389836855775840822719678533376883030929209752909248937768242

**Reply by sshmatrix** [reply]
Got it; Reverse Record then. Since `name` is already implemented in the new Public Resolver then we shouldn't need to CCIP-Read it separately, I think. It should work similar to `addr` or `pubkey` or whatever. Or, am I missing some deeper catch in `name` implementation?

**Reply by 0xc0de4c0ffee** [reply]
yup, we've to understand how new stuffs work & test all properly, 
there's no official docs/ENSIP for all changes pushed with namewrapper YET, = we've to read the codes 🤣

**Reply by 0xc0de4c0ffee** [reply]
As suggested before we should leave reverse records/resolvers to on-chain version.

adding reverse mapping for 0xaddr to domain.eth on off-chain will require full extra setup for node 0zaddr.addr.reverse records. 
owner have to set IPNS contenthash for both domain.eth node and reverse node record to add such records.
It's doable for some extra gas but it's better if reverse record is set on official resolver as it's mostly on-time records.

We can reuse `name(...)` function in off-chain to store plan name for domain.eth, e.g. "My Domain"
